src/services/contract-service.js

```diff
@@ -276,8 +277,15 @@ class ContractService {
       method
         .send(opts)
         .on('receipt', resolve)
-        .on('confirmation', confirmationCallback)
-        .on('transactionHash', transactionHashCallback)
+        //.on('confirmation', confirmationCallback)
```

Comment:
Just curious: we can't subscribe to this at all? It's what results in an exception?

Reply:
I just reworked this part since I wasn't happy with canceling functionality that worked in some cases. Working on a revamp of this. My idea is: 
- keep the on("confirmation") logic in 
- also start pooling the fallback function
- if we receive the event in on("confirmation") cancel the fallback function pooling, since we don't need it